src/compute/src/render/mod.rs

```diff
@@ -544,19 +544,15 @@ where
                 Id::Global(idx_id)
             )
         });
-        let arrangement = bundle.arrangement(&idx.key);
 
-        // Set up probes to notify on index frontier advancement.
-        if let Some(arr) = &arrangement {
-            let (collection, _) = arr.as_collection();
-            let stream = collection.inner;
-            stream.probe_notify_with(probes.clone());
-        }
-
-        compute_state.flow_control_probes.insert(idx_id, probes);
+        compute_state
+            .flow_control_probes
+            .insert(idx_id, probes.clone());
 
-        match arrangement {
+        match bundle.arrangement(&idx.key) {
             Some(ArrangementFlavor::Local(oks, errs)) => {
+                // Set up probes to notify on index frontier advancement.
+                oks.stream.probe_notify_with(probes);
```

Comment:
We don't attach probes when the arrangement has `Trace` flavor. Why don't we care about that case?

Reply:
I think in this case there should have been a probe attached already, but let's check this is true!

Reply:
I think this is correct because all traces need to be exported at some point, and this is where we attach the probes. Importing them later doesn't modify the set of probes attached to a specific trace. Only log sources don't have a probe attached to them, but that's OK because they don't read from external data.